### PR TITLE
Introduce pnpm workspace

### PR DESCRIPTION
**Pull request by LEAFERx**
# Goals of PR

Core changes:

- Introduce pnpm workspace

Notes:

- Dependency #99

### CODE REVIEW

File: .github/actions/node/action.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: 'Setup Node Workspace'
```
Comment (ujvl) [inline .github/actions/node/action.yml:1]: nit: rename file to `setup.yml`?

File: contracts/deploy/l1/00_deploy_SequencerInbox.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types";
```
Comment (ujvl) [inline contracts/deploy/l1/00_deploy_SequencerInbox.ts:1]: should we move `deploy` into `scripts`

nit: also never understood why these files need to start with a number, is it to enforce deployment ordering

File: contracts/hardhat.config.ts [hunk 3]
```diff
@@ -71,19 +100,28 @@ const config: HardhatUserConfig = {
...
   },
   networks: {
     mainnet: createConfig("mainnet"),
     sepolia: createConfig("sepolia"),
     goerli: createConfig("goerli"),
     gnosis: createConfig("gnosis"),
     chiado: createConfig("chiado"),
+    localhost: createConfig("localhost"),
+    specularLocalDev: createConfig("specularLocalDev"),
+    specularDev: createConfig("specularDev"),
```
Comment (ujvl) [inline contracts/hardhat.config.ts:124]: can you add comments explaining these

**Review by ujvl — APPROVED** [review]
minor comments but otherwise lgtm

File: package.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
```
[the remark this answers is not in the document]
Comment (LEAFERx) [reply package.json:1]: this is the package.json for the entire workspace, which can help setup some utilities across different projects like husky, linting, etc

File: contracts/deploy/l1/00_deploy_SequencerInbox.ts (hunk 2 above)
Comment (LEAFERx) [reply contracts/deploy/l1/00_deploy_SequencerInbox.ts:1]: scripts should contains files that are invoked directly by users, but deploy is invoked by hardhat-deploy plugin;
yes the number is to enforce deployment ordering

File: contracts/hardhat.config.ts (hunk 3 above)
Comment (LEAFERx) [reply contracts/hardhat.config.ts:124]: localdev is the specular network while localhost is the local l1; will add comments

File: .github/actions/node/action.yml (hunk 1 above)
Comment (LEAFERx) [reply .github/actions/node/action.yml:1]: can't, github requires an action to be named as `action.yml`

**Review by ujvl — APPROVED** [review]

**Pull request merged**